docs/spec/metricsets/metricset.json

```diff
@@ -22,11 +20,61 @@
                     },
                     "additionalProperties": false
                 },
+                "span": {
+                    "type": [
+                        "object",
+                        "null"
+                    ],
+                    "properties": {
+                        "sub_type": {
```

Comment:
```suggestion
                        "subtype": {
```

Reply:
Maybe we should continue to use span_subtype.json etc. though, but nested under the `span` object? `span.json` uses it; I think it would be great if we could maintain a single definition.

Reply:
The main reason to get rid of the sub-schema json files here was easier readability. The `span_type.json` and `transaction_type.json` defined `"type": "string"`, which is aligned with requirements for `span.json` and `transaction.json` but when reusing for `metricsets` the type needs to also include `null` as the fields are optional. Referencing the files and changing the types of the deep nested attributes can easily lead to errors (as I've seen on `service.*` related fields just recently) and makes the schemar harder to read. Since all of this is subject to change soon anyways, I'd go with some duplication and easier readability right now.

Reply:
Solid points. Sounds good, thanks!